analysis/vendor/res_syntax/res_outcome_printer.ml

```diff
@@ -322,9 +322,7 @@ let rec printOutTypeDoc (outType : Outcometree.out_type) =
       ]
 
 and printOutArrowType ~uncurried typ =
-  let uncurried =
-    if !Config.uncurried <> Legacy then not uncurried else uncurried
-  in
+  let uncurried = Res_uncurried.getDotted ~uncurried !Config.uncurried in
```

Comment:
PR:ed to the compiler here: https://github.com/rescript-lang/rescript-compiler/pull/6353

Reply:
Does that fix the issue?

Reply:
It fixes the printing issues in this repo, yes. There were dots in lots of places there shouldn't be in uncurried mode before this.